tests/test_pipeline_media.py

```diff
@@ -124,8 +124,8 @@ def test_should_remove_req_res_references_before_caching_the_results(self):
         # Simulate the Media Pipeline behavior to produce a Twisted Failure
         try:
             # Simulate a Twisted inline callback returning a Response
-            # The returnValue method raises an exception encapsulating the value
-            returnValue(response)
+            # The returned Response raises an exception
```

Comment:
It doesn't. `defer.returnValue` indeed raises an exception so it should either keep the test as is or change it, this depends on the test purpose. It may be related to the comment in MediaPipeline, or maybe not.

Reply:
+1 for rewriting the test. I think it'd be good to clean [this block](https://github.com/scrapy/scrapy/blob/2.0.1/scrapy/pipelines/media.py#L141-L162) up, since the mentioned `_DefGen_Return` exception would not be in play anymore.